### ISSUE TITLE
Can it compile to a dll/so?

### ISSUE THREAD

**Issue by Zireael07** [titzer/virgil#97]
Title.

**Comment by titzer** [comment]
Currently no. The Virgil compiler always produces a statically linked executable. Would your use case need linking a .so?

**Comment by Zireael07** [comment]
Yes (it's for a preexisting game engine that can use external dll's for scripting languages support. No hurry though, as I don't expect their docs/API to be finalized in a couple months from now

**Comment by titzer** [comment]
I don't have the bandwidth on that atm, but off the top of my head, it would require emitting an additional section for the dynamic linker and some internal compiler work to deal with addresses that aren't fixed.
